### ISSUE TITLE
PHP Warning: Undefined variable $changed_stock" in includes/class-wc-ajax.php on line 1336

### ISSUE THREAD

**Issue by twwn** [woothemes/woocommerce#32717]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Seen in the log today:

>PHP Warning:  Undefined variable $changed_stock in /srv/wp-content/plugins/woocommerce/includes/[class-wc-ajax.php on line 1336](https://github.com/woocommerce/woocommerce/blob/6.4.1/plugins/woocommerce/includes/class-wc-ajax.php#L1336)

Unfortunately I don't know what caused it, but looking at the affected line & above, that variable is [indeed only defined](https://github.com/woocommerce/woocommerce/blob/6.4.1/plugins/woocommerce/includes/class-wc-ajax.php#L1307) after two `if` checks – rather than being defined ahead or covered by an `else` branch.

I take it's the second check that caused the problem here as otherwise the first two parameters `$item_id` & `$item` would have been undefined as well. Seems those two could use a default, too.

### Expected behavior

Nobody complained, so seems harmless.

### Actual behavior

No entry in the debug log.

### Steps to reproduce

1. Use WooCommerce
2. Don't know, unfortunately

### WordPress Environment

`
### WordPress Environment ###

WC Version: 6.4.1
REST API Version: ✔ 6.4.1
WC Blocks Version: ✔ 7.2.2
Action Scheduler Version: ✔ 3.4.0
WC Admin Version: ✔ 3.3.2
Log Directory Writable: ✔
WP Version: 5.9.3
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 8.0.17
PHP Post Max Size: 120 MB
PHP Time Limit: 0
PHP Max Input Vars: 5000
cURL Version: 7.61.1
OpenSSL/1.1.1k

SUHOSIN Installed: –
MySQL Version: 10.4.24-MariaDB
Max Upload Size: 100 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.4.1
WC Database Prefix: wp_

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (50) ###

Complianz | GDPR/CCPA Cookie Consent: by Really Simple Plugins – 6.1.5
DHL for WooCommerce: by DHL – 2.8.12
WooCommerce Admin: by WooCommerce – 3.3.2
German Market: by MarketPress – 3.14.1
WooCommerce: by Automattic – 6.4.1
Yoast SEO: by Team Yoast – 18.6

### Dropin Plugins (1) ###

advanced-cache.php: advanced-cache.php

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
gift-card (gift-card)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #4390616 - /neu-im-sortiment/
Cart: #20 - /warenkorb/
Checkout: #21 - /kasse/
My account: #22 - /mein-konto/
Terms and conditions: #40 - /agb/

### Theme ###

Name: Divi Jewellery Shop
Version: 1.0.1
Author URL: http://divi-childthemes.com
Child Theme: ✔
Parent Theme Name: Divi
Parent Theme Version: 4.17.1
Parent Theme Author URL: http://www.elegantthemes.com
WooCommerce Support: ✔

### Action Scheduler ###

Complete: 4,269
Oldest: 2022-04-02 10:22:28 +0200
Newest: 2022-04-21 14:29:49 +0200

Pending: 6
Oldest: 2022-04-21 15:29:49 +0200
Newest: 2022-04-28 12:37:51 +0200

### Status report information ###

Generated at: 2022-04-21 14:31:24 +02:00
`

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Comment by froger-me** [comment]
I get this warning when deleting a fee item from the order interface.

## Reproducing:

### Pre-requisite:
An completed order with a fee (in my case: a custom fee added at checkout time).

- change the order status to pending payment and save
- click the delete icon associated to the fee
- view logs and see the warning

**Comment by twwn** [comment]
This is still occurring — and as explained, with code path obviously missing the variable definitions, the bug is entirely apparent.

Should this not be fixed?

**Comment by jorgeatorres** [comment]
Thanks everyone for the report and apologize for the delay in responding. We were able to reproduce and will address in a future version.

**Comment by aleferri** [comment]
Just fix putting $changed_stock to false before the check, do you really need a PR for a one line fix?

`

      $changed_stock = false; // False = NO STOCK CHANGED, per current implementation of wc_maybe_adjust_line_item_product_stock(), WC 7.4.1

      if ( ! empty( $order_item_ids ) ) {
      
	      foreach ( $order_item_ids as $item_id ) {
		      $item_id = absint( $item_id );
		      $item    = $order->get_item( $item_id );
      
		      // Before deleting the item, adjust any stock values already reduced.
		      if ( $item->is_type( 'line_item' ) ) {
			      $changed_stock = wc_maybe_adjust_line_item_product_stock( $item, 0 );
      
			      if ( $changed_stock && ! is_wp_error( $changed_stock ) ) {
					      /* translators: %1$s: item name %2$s: stock change */
					      $order->add_order_note( sprintf( __( 'Deleted %1$s and adjusted stock (%2$s)', 'woocommerce' ), $item->get_name(), $changed_stock['from'] . '&rarr;' . $changed_stock['to'] ), false, true );
			      } else {
			                      /* translators: %s item name. */
					      $order->add_order_note( sprintf( __( 'Deleted %s', 'woocommerce' ), $item->get_name() ), false, true );
			      }
		      }
      
		      wc_delete_order_item( $item_id );
	      }
      }

`
